### ISSUE TITLE
TypeError: Dict key must be str - when launching parllama

### ISSUE THREAD

**Issue by PureStupid** [paulrobello/parllama#29]
**Describe the bug**
I'm not sure whether this is an issue on my end: When trying to launch parllama, the app's UI is shown to be loading momentarily, before the following traceback is received:

```
╭─────────────────────────────────────────── Traceback (most recent call last) ────────────────────────────────────────────╮
│ C:\Users\Enlet\parllama\.venv\Lib\site-packages\textual\worker.py:368 in _run                                            │
│                                                                                                                          │
│   365 │   │   self.state = WorkerState.RUNNING                                                                           │
│   366 │   │   app.log.worker(self)                                                                                       │
│   367 │   │   try:                                                                                                       │
│ ❱ 368 │   │   │   self._result = await self.run()                                                                        │
│   369 │   │   except asyncio.CancelledError as error:                                                                    │
│   370 │   │   │   self.state = WorkerState.CANCELLED                                                                     │
│   371 │   │   │   self._error = error                                                                                    │
│                                                                                                                          │
│ ╭─────────────────────────────────────────────── locals ────────────────────────────────────────────────╮                │
│ │           app = ParLlamaApp(title='PAR LLAMA', classes={'-dark-mode'})                                │                │
│ │         error = TypeError('Dict key must be str')                                                     │                │
│ │          self = <Worker ERROR name='refresh_provider_models' description='refresh_provider_models()'> │                │
│ │     Traceback = <class 'rich.traceback.Traceback'>                                                    │                │
│ │ worker_failed = WorkerFailed("Worker raised exception: TypeError('Dict key must be str')")            │                │
│ ╰───────────────────────────────────────────────────────────────────────────────────────────────────────╯                │
│                                                                                                                          │
│ C:\Users\Enlet\parllama\.venv\Lib\site-packages\textual\worker.py:352 in run                                             │
│                                                                                                                          │
│   349 │   │   Returns:                                                                                                   │
│   350 │   │   │   Return value of the work.                                                                              │
│   351 │   │   """                                                                                                        │
│ ❱ 352 │   │   return await (                                                                                             │
│   353 │   │   │   self._run_threaded() if self._thread_worker else self._run_async()                                     │
│   354 │   │   )                                                                                                          │
│   355                                                                                                                    │
│                                                                                                                          │
│ ╭─────────────────────────────────────────── locals ───────────────────────────────────────────╮                         │
│ │ self = <Worker ERROR name='refresh_provider_models' description='refresh_provider_models()'> │                         │
│ ╰──────────────────────────────────────────────────────────────────────────────────────────────╯                         │
│                                                                                                                          │
│ C:\Users\Enlet\parllama\.venv\Lib\site-packages\textual\worker.py:337 in _run_async                                      │
│                                                                                                                          │
│   334 │   │   │   or hasattr(self._work, "func")                                                                         │
│   335 │   │   │   and inspect.iscoroutinefunction(self._work.func)                                                       │
│   336 │   │   ):                                                                                                         │
│ ❱ 337 │   │   │   return await self._work()                                                                              │
│   338 │   │   elif inspect.isawaitable(self._work):                                                                      │
│   339 │   │   │   return await self._work                                                                                │
│   340 │   │   elif callable(self._work):                                                                                 │
│                                                                                                                          │
│ ╭─────────────────────────────────────────── locals ───────────────────────────────────────────╮                         │
│ │ self = <Worker ERROR name='refresh_provider_models' description='refresh_provider_models()'> │                         │
│ ╰──────────────────────────────────────────────────────────────────────────────────────────────╯                         │
│                                                                                                                          │
│ C:\Users\Enlet\parllama\src\parllama\app.py:799 in refresh_provider_models                                               │
│                                                                                                                          │
│   796 │   @work(exclusive=True)                                                                                          │
│   797 │   async def refresh_provider_models(self) -> None:                                                               │
│   798 │   │   """Refresh provider models"""                                                                              │
│ ❱ 799 │   │   provider_manager.refresh_models()                                                                          │
│   800 │                                                                                                                  │
│   801 │   @on(RefreshProviderModelsRequested)                                                                            │
│   802 │   def on_refresh_provider_models(self, event: RefreshProviderModelsRequested) -> None:                           │
│                                                                                                                          │
│ ╭─────────────────────────── locals ────────────────────────────╮                                                        │
│ │ self = ParLlamaApp(title='PAR LLAMA', classes={'-dark-mode'}) │                                                        │
│ ╰───────────────────────────────────────────────────────────────╯                                                        │
│                                                                                                                          │
│ C:\Users\Enlet\parllama\src\parllama\provider_manager.py:107 in refresh_models                                           │
│                                                                                                                          │
│   104 │   │   │   except Exception as e:  # pylint: disable=broad-exception-caught                                       │
│   105 │   │   │   │   print(f"Error: {e}")                                                                               │
│   106 │   │   │   │   continue                                                                                           │
│ ❱ 107 │   │   self.save_models()                                                                                         │
│   108 │                                                                                                                  │
│   109 │   # pylint: disable=too-many-return-statements, too-many-branches                                                │
│   110 │   @staticmethod                                                                                                  │
│                                                                                                                          │
│ ╭───────────────── locals ──────────────────╮                                                                            │
│ │ new_list = []                             │                                                                            │
│ │        p = <LlmProvider.GOOGLE: 'Google'> │                                                                            │
│ │     self = ProviderManager()              │                                                                            │
│ ╰───────────────────────────────────────────╯                                                                            │
│                                                                                                                          │
│ C:\Users\Enlet\parllama\src\parllama\provider_manager.py:130 in save_models                                              │
│                                                                                                                          │
│   127 │                                                                                                                  │
│   128 │   def save_models(self):                                                                                         │
│   129 │   │   """Save the models."""                                                                                     │
│ ❱ 130 │   │   self.cache_file.write_bytes(json.dumps(self.provider_models))                                              │
│   131 │                                                                                                                  │
│   132 │   def load_models(self, refresh: bool = False) -> None:                                                          │
│   133 │   │   """Load the models."""                                                                                     │
│                                                                                                                          │
│ ╭───────── locals ─────────╮                                                                                             │
│ │ self = ProviderManager() │                                                                                             │
│ ╰──────────────────────────╯                                                                                             │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
TypeError: Dict key must be str
```
Reinstalling with `make resetup`, and clearing the `.parllama` folder did not resolve it.
I have also installed using pipx, to no avail.

**To Reproduce**
1. Attempt to launch parllama.

**Expected behavior**
Parllama should launch as per normal.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Windows 11
 - Browser: NA
 - Version 10.0.22631

**Additional context**
# Textual Diagnostics

## Versions

| Name    | Value  |
|---------|--------|
| Textual | 0.79.0 |
| Rich    | 13.8.1 |

## Python

| Name           | Value                                            |
|----------------|--------------------------------------------------|
| Version        | 3.12.4                                           |
| Implementation | CPython                                          |
| Compiler       | MSC v.1929 64 bit (AMD64)                        |
| Executable     | C:\Users\Enlet\parllama\.venv\Scripts\python.exe |

## Operating System

| Name    | Value      |
|---------|------------|
| System  | Windows    |
| Release | 11         |
| Version | 10.0.22631 |

## Terminal

| Name                 | Value            |
|----------------------|------------------|
| Terminal Application | Windows Terminal |
| TERM                 | *Not set*        |
| COLORTERM            | *Not set*        |
| FORCE_COLOR          | *Not set*        |
| NO_COLOR             | *Not set*        |

## Rich Console options

| Name           | Value                |
|----------------|----------------------|
| size           | width=124, height=33 |
| legacy_windows | False                |
| min_width      | 1                    |
| max_width      | 124                  |
| is_terminal    | True                 |
| encoding       | utf-8                |
| max_height     | 33                   |
| justify        | None                 |
| overflow       | None                 |
| no_wrap        | False                |
| highlight      | None                 |
| markup         | None                 |
| height         | None                 |

**Comment by paulrobello** [comment]
I can reproduce.
I hope to have a fix released within 24 hours. Probably less.

**Comment by abdessalaam** [comment]
I have the same error.

**Comment by paulrobello** [comment]
Just pushed v0.3.10 which should fix the issue.

**Comment by abdessalaam** [comment]
> Just pushed v0.3.10 which should fix the issue.

It works, very nice! Thank you.

**Comment by PureStupid** [comment]
I can confirm it has now been fixed in #30 . Thanks!
